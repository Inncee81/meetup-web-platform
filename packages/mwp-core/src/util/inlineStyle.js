// @flow

import React from 'react';
import { getStyles } from 'simple-universal-style-loader';

/**
 * Returns inline style tags from CSS accumulated by
 * `simple-universal-style-loader`. This allows us to
 * render styles from CSS modules on the server.
 *
 * @returns {Array} - array of style tags
 */
export function getInlineStyleTags(): Array<React$Element<'style'>> {
	return getStyles() && getStyles()
		.map(style =>
			<style
				type="text/css"
				key={style.id}
			>
				{style.parts.map(part => part.css)}
			</style>
		);
};
