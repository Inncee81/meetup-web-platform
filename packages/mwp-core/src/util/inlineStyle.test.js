import { getInlineStyleTags } from './inlineStyle.js';

// `simple-universal-style-loader` reades from `global`,
// so we must mock the data directly in `global.__styles__`
const MOCK_GLOBAL_STYLES = [
	{
		id: 666,
		parts: [
			{
				css: '.foo { overflow: inherit; }',
				media: ''
			}
		]
	}
];

describe('getInlineStyleTags', () => {
	it('matches snapshot', () => {
		global.__styles__ = MOCK_GLOBAL_STYLES;
		const result = getInlineStyleTags()[0];
		expect(result).toMatchSnapshot();
		global.__styles__ = null;
	});
	it('does not throw when global styles are not populated', () => {
		expect(() => getInlineStyleTags()).not.toThrow();
	});
});
