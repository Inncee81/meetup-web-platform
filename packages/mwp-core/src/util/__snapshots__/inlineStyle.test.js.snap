// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getInlineStyleTags matches snapshot 1`] = `
<style
  type="text/css"
>
  .foo { overflow: inherit; }
</style>
`;
